src/Symfony/Bundle/FrameworkBundle/Test/TestContainer.php

```diff
@@ -69,7 +68,20 @@ public function setParameter(string $name, mixed $value)
 
     public function set(string $id, mixed $service)
     {
-        $this->getPublicContainer()->set($id, $service);
+        $container = $this->getPublicContainer();
+        $renamedId = $this->renamedIds[$id] ?? $id;
+
+        try {
+            $container->set($renamedId, $service);
+        } catch (InvalidArgumentException $e) {
+            if (!str_starts_with($e->getMessage(), "The \"$renamedId\" service is private")) {
```

Comment:
Couldn't we use a magic number as an exception code instead to bulletproof this?

Reply:
we do this already and this is tested so I wouldn't mind